spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -431,6 +447,32 @@ def sample_allocation(obj)
           expect(heap_samples.map(&:labels)).to all(match(hash_including(:'gc gen age' => be_a(Integer).and(be >= 0))))
         end
 
+        it 'include accurate object sizes' do
+          string_sample = heap_samples.find { |s| s.labels[:'allocation class'] == 'String' }
+          expect(string_sample.values[:'heap-live-size']).to be_between(
+            # 18 UTF-8 characters at minimum.
+            (18 * 2) * sample_rate,
+            # Add some extra padding (per char and object-wide) for extra data.
+            (18 * 4 + 40) * sample_rate
+          )
```

Comment:
So at first pass I was a bit confused here with the math -- why 18*2 (if utf-8 represents ascii chars with 1 byte), etc. I realized actually the size of `a_string` is 40, because it fits within a single slot.

A quick experiment with `ObjectSpace.memsize_of` says you can have 23 ascii chars in there, which is a bit like an array -- 24 bytes is the same 3 words that an array can fit.

All that to say: maybe instead of asserting directly on values, we could rely on Ruby's value to be accurate and directly use it in the equation?

```suggestion
          expect(string_sample.values[:'heap-live-size'])
            .to eq(ObjectSpace.memsize_of(a_string) * sample_rate) 
```

(Same for the others?)

Reply:
Hmm yeah I was reluctant to use `memsize_of` directly here because that gives us no notion of whether the values returned by `memsize_of` are "reasonable" (or if they'd all be fixed to, for example, 40 due to some issue/bug).

I do agree that the current range-checking has quite arbitrary (and potentially flaky) bounds though so maybe better to do what you suggest and leave correctness checks for some other more "aggregated"/E2E test.